pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -0,0 +1,144 @@
+	return []string{event.AnyNetworkPlugin}
+}
+
+func (h *mtuHandler) GetName() string {
+	return "MTU handler"
+}
+
+func (h *mtuHandler) Init() error {
+	var err error
+	h.ipt, err = iptables.New()
+	if err != nil {
+		return fmt.Errorf("error initializing iptables: %v", err)
+	}
+
+	if err := iptables.CreateChainIfNotExists(h.ipt, mangleTable, subMarinerPostRoutingChain); err != nil {
+		return fmt.Errorf("error creating iptables chain %s: %v", subMarinerPostRoutingChain, err)
+	}
+
+	forwardToSubMarinerPostRoutingChain := []string{"-j", subMarinerPostRoutingChain}
+
+	if err := iptables.PrependUnique(h.ipt, mangleTable, postRoutingChain, forwardToSubMarinerPostRoutingChain); err != nil {
+		return fmt.Errorf("error inserting iptables rule %q: %v",
+			strings.Join(forwardToSubMarinerPostRoutingChain, " "), err)
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	subnets := extractSubnets(endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.addClampMssToPathMTU(subnet)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
+	subnets := extractSubnets(endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.removeClampMssToPathMTU(subnet)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) addClampMssToPathMTU(subnets string) error {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := h.ipt.AppendUnique(mangleTable, subMarinerPostRoutingChain, ruleSpecSource...); err != nil {
+		return fmt.Errorf("error appending iptables rule \"%s\": %v", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := h.ipt.AppendUnique(mangleTable, subMarinerPostRoutingChain, ruleSpecDest...); err != nil {
+		return fmt.Errorf("error appending iptables rule \"%s\": %v", strings.Join(ruleSpecDest, " "), err)
+	}
+
+	return nil
+}
+
+func (h *mtuHandler) removeClampMssToPathMTU(subnets string) error {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := h.ipt.Delete(mangleTable, subMarinerPostRoutingChain, ruleSpecSource...); err != nil {
+		return fmt.Errorf("error deleting iptables rule \"%s\": %v", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := h.ipt.Delete(mangleTable, subMarinerPostRoutingChain, ruleSpecDest...); err != nil {
+		return fmt.Errorf("error deleting iptables rule \"%s\": %v", strings.Join(ruleSpecDest, " "), err)
+	}
```

Comment:
Let's print those error if they happen but let's not return fmt.Errorf

only log.Errorf(

In this way the error is soft, we log it, we can continue attempting to delete the rules (for example if there was an issue creating them initially).